cpp/tests/test_sde_seirvv.cpp

```diff
@@ -0,0 +1,234 @@
+    parameters.set<mio::sseirvv::TimeInfectedV2>(0);
+    EXPECT_EQ(parameters.check_constraints(), 1);
+
+    parameters.set<mio::sseirvv::TimeInfectedV2>(6);
+    parameters.set<mio::sseirvv::TimeExposedV1>(0); 
+    EXPECT_EQ(parameters.check_constraints(), 1);
+
+    parameters.set<mio::sseirvv::TimeExposedV1>(6);
+    parameters.set<mio::sseirvv::TimeExposedV2>(0);
+    EXPECT_EQ(parameters.check_constraints(), 1);
+
+    parameters.set<mio::sseirvv::TimeExposedV2>(6);
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(10.);
+    EXPECT_EQ(parameters.check_constraints(), 1);
+
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(0.04);
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV2>(10.);
+    EXPECT_EQ(parameters.check_constraints(), 1);
+    mio::set_log_level(mio::LogLevel::warn);
+}
+
+TEST(TestSdeSeirvv, apply_constraints_parameters)
+{
+    const double tol_times = 1e-1;
+    mio::sseirvv::Model::ParameterSet parameters;
+    parameters.set<mio::sseirvv::TimeInfectedV1>(6);
+    parameters.set<mio::sseirvv::TimeInfectedV2>(6);
+    parameters.set<mio::sseirvv::TimeExposedV1>(6);
+    parameters.set<mio::sseirvv::TimeExposedV2>(6);
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(0.04);
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV2>(0.04);
+    parameters.get<mio::sseirvv::ContactPatterns>().get_baseline()(0, 0) = 10;
+
+    EXPECT_EQ(parameters.apply_constraints(), 0);
+
+    mio::set_log_level(mio::LogLevel::off);
+
+    parameters.set<mio::sseirvv::TimeInfectedV1>(-2.5);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_EQ(parameters.get<mio::sseirvv::TimeInfectedV1>(), tol_times);
+
+    parameters.set<mio::sseirvv::TimeInfectedV2>(-2.5);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_EQ(parameters.get<mio::sseirvv::TimeInfectedV2>(), tol_times);
+
+    parameters.set<mio::sseirvv::TimeExposedV1>(-2.5);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_EQ(parameters.get<mio::sseirvv::TimeExposedV1>(), tol_times);
+
+    parameters.set<mio::sseirvv::TimeExposedV2>(-2.5);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_EQ(parameters.get<mio::sseirvv::TimeExposedV2>(), tol_times);
+
+    parameters.set<mio::sseirvv::TimeExposedV2>(-2.5);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_EQ(parameters.get<mio::sseirvv::TimeExposedV2>(), tol_times);
+
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV1>(10.);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_NEAR(parameters.get<mio::sseirvv::TransmissionProbabilityOnContactV1>(), 0.0, 1e-14);
+
+    parameters.set<mio::sseirvv::TransmissionProbabilityOnContactV2>(10.);
+    EXPECT_EQ(parameters.apply_constraints(), 1);
+    EXPECT_NEAR(parameters.get<mio::sseirvv::TransmissionProbabilityOnContactV2>(), 0.0, 1e-14);
+    mio::set_log_level(mio::LogLevel::warn);
+}
```

Comment:
```suggestion
}

```
Small detail, but missing endline.